server/controllers/ArticleController.js

```diff
@@ -0,0 +1,254 @@
+import { Boolify } from 'node-boolify';
+import db from '../models';
+import ArticleHelper from '../helpers/ArticleHelper';
+import Util from '../helpers/Util';
+import TagHelper from '../helpers/TagHelper';
+import response from '../helpers/response';
+
+
+const { Article, Tag, User } = db;
+
+/**
+ * @class ArticleController
+ */
+class ArticleController {
+  /**
+   * @static
+   * @description a function that handles addition of new article to the db
+   * @param {object} req HTTP request object
+   * @param {object} res HTTP response object
+   * @returns {object} api route response with created article info
+   */
+  static async create(req, res) {
+    try {
+      const { id } = req.user;
+      const {
+        title,
+        content,
+        isPublished,
+        banner,
+        tagsList,
+      } = req.body;
+      const tagsArray = tagsList ? Util.createArrayOfStrings(tagsList) : '';
+
+      const articleData = {
+        userId: id,
+        slug: ArticleHelper.generateArticleSlug(title),
+        title,
+        content,
+        banner: banner || 'https://unsplash.com/photos/Q7wDdmgCBFg',
+        tagsList: tagsArray,
+        isPublished: Boolify(isPublished) || true,
+        isReported: false,
+      };
+
+      const article = await Article.create(articleData);
+
+      if (article) {
+        const {
+          createdAt,
+          updatedAt
+        } = article;
+        await TagHelper.findOrAddTag(article.id, tagsArray);
+        articleData.createdAt = Util.formatDate(createdAt);
+        articleData.updatedAt = Util.formatDate(updatedAt);
+
+        return response(
+          res, 201, 'success',
+          'New article has been successfully created',
+          null, article
+        );
+      }
+    } catch (error) {
+      return response(
```

Comment:
I think we need to have access to the thrown error here. I understand the need to have a nice error message sent in the response but knowing exactly the error we are getting is importanta. I think we should rethink this response function and generally our error handling architecture.

Reply:
I agree with you @jesseinit, You raised a valid point. I'll suggest we use libs like winston or any logger libs to log errors. The team needs to discuss on this.

Reply:
An error should not be thrown to the user, I think logging errors instead should be better.